tests/integration/weight_test.go

```diff
@@ -41,4 +35,30 @@ func weightTests(t *testing.T, when spec.G, it spec.S) {
 			assert.GreaterOrEqual(t, responses["Hello World v3"], 1, "The application did not return enough responses from the staged service. which has slightly less weight than the service.")
 		})
 	}, spec.Parallel())
+
+	when("a staged service is promoted", func() {
+		it.Before(func() {
+			service.Create(t, "--weight", "100", "ibuildthecloud/demo:v1")
+			stagedService = service.Stage("ibuildthecloud/demo:v3", "v3")
+			stagedService.Promote()
+		})
+
+		it("should retain all revision endpoints with an app endpoint pointing to the new version", func() {
+			assert.Equal(t, "Hello World", service.GetEndpointResponse())
+			assert.Equal(t, "Hello World v3", stagedService.GetEndpointResponse())
+			assert.Equal(t, 0, service.GetCurrentWeight())
+			assert.Equal(t, 100, stagedService.GetCurrentWeight())
+			//assert.Equal(t, 0, service.GetKubeCurrentWeight())
```

Comment:
Should just delete these I think. We would need to do a full calc with kubectl to do it and that logic might change in rio which makes anything in tests brittle, so I don't think they are valuable. You can do in follow up PR.